adafruit_esp32spi/adafruit_esp32spi_socket.py

```diff
@@ -113,6 +113,19 @@ def recv_into(self, buffer, nbytes: int = 0):
         num_to_read = len(buffer) if nbytes == 0 else nbytes
         num_read = 0
         while num_to_read > 0:
+            # we might have read socket data into the self._buffer with:
+            # esp32spi_wsgiserver: socket_readline
```

Comment:
We discarded `esp32spi_wsigiserver`. Are there other circumstances where is could happen?

Reply:
I haven't checked any other libraries where `_buffer` is used, only https://github.com/adafruit/Adafruit_CircuitPython_WSGI/blob/main/adafruit_wsgi/esp32spi_wsgiserver.py

But i think is kinda of expected behavior and reasonable to use the `_buffer`, read as much as possible from the socket, because is not performant to read one bit at a time to find EOL for example.

This implementation also made into Adafruit_CircuitPython_Wiznet5k,  adafruit_wiznet5k_socket `_readline` and adafruit_wiznet5k_wsgiserver so maybe is spread.